<?xml version="1.0" encoding="UTF-8"?>

<launch>
    <arg name="sequence" default="sequence_1" />
    <arg name="bag_file" default="/home/voliro/data/recordings_with_state_estimation/hg_outdoors_1_new.bag"/>
    
    <arg name="eval_folder" default="/home/voliro/data/" />
    <arg name="experiment_name" default="results_HG_d_floor_1" />
    <arg name="config_file" default="$(find lidar_motion_detection)/config/ros_params_MultiClassDynamicObjectsStairs.yaml" />
    <arg name="player_rate" default="0.4" />
    <param name="use_sim_time" value="true" />
    <node type="rviz" name="rviz_hauptgebaeude" pkg="rviz"
     args="-d $(find lidar_motion_detection)/rviz/recordings.rviz" output="log"/>

    <node name="player" pkg="rosbag" type="play" output="log"
     args="--quiet --rate $(arg player_rate) --clock $(arg bag_file)" required="true" />

    <node name="lidar_undistortion" pkg="lidar_undistortion" type="lidar_undistortion_node" output="log" >
        <remap from="pointcloud" to="/os_cloud_node/points" />
        <param name="odom_frame_id" value="map" />
        <param name="lidar_frame_id" value="os_lidar" />
    </node>

    <node name="vis_node" pkg="lidar_motion_detection" type="motion_detector" output="screen"
     args="-alsologtostderr" clear_params="true" >

        <remap from="pointcloud_intercepted" to="/lidar_undistortion/pointcloud_corrected"/>

        <rosparam command="load" file="$(arg config_file)" />
        <param name="config_file" value="$(arg config_file)" />
        <param name="bag_file" value="$(arg bag_file)" />
        <param name="sequence_name" value="$(arg sequence)" />
        <param name="eval_folder" value="$(arg eval_folder)" />
        <param name="experiment_name" value="$(arg experiment_name)" />
    </node>

</launch>
